### PR TITLE
Add code to retrieve ticket Tags and also Create comments

### PR DESCRIPTION
**Pull request by martyzz1**

### CODE REVIEW

File: assembla/api.py [hunk 1]
```diff
@@ -379,6 +380,19 @@ def users(self, extra_params=None):
...
+        All Tags in this Space
+        """
+        return self.api._get_json(
+            Tag,
+            space=self,
+            rel_path=self._build_rel_path('tags'),
+            extra_params=extra_params,
+        )
+
+    @assembla_filter
```
Comment (markfinger) [inline assembla/api.py:394]: Can you remove the doubled decorator?

**Comment by markfinger** [comment]
Sorry, I'm a bit confused by the namespace system that you're introducing. Can you summarise the changes you're making and why?

File: assembla/api.py [hunk 2]
```diff
@@ -466,6 +505,10 @@ def comments(self, extra_params=None):
         if extra_params:
             params.update(extra_params)
 
+        rel_path=self.space._build_rel_path(
```
Comment (markfinger) [inline assembla/api.py:508]: Is this doing anything?

File: assembla/api.py [hunk 3]
```diff
@@ -508,8 +551,45 @@ def delete(self):
...
+        if self.get('id'):  # Modifying an existing ticket
+            method = self.ticket.api._put_json
+        else:  # Creating a new ticket
+            method = self.ticket.api._post_json
+
+        return method(
+            self,
+            space=self.ticket.space,
+            rel_path=self.ticket.space._build_rel_path(
+                    'tickets/%s/ticket_comments' % self.ticket['number']
```
Comment (markfinger) [inline assembla/api.py:585]: Can you fix the indentation on this line

File: assembla/api.py (hunk 2 above)
Comment (markfinger) [inline assembla/api.py:508]: It looks like it's being overridden below.

**Comment by markfinger** [comment]
Hi,

I've merged in your code relating to the Tags, as it's all pretty self-explanatory. I added some docs as well.

I'll close the pull request, but I'm happy to merge another one in, if you can explain the changes. I'm a bit reluctant to make changes to the core parts of this package, as I don't really use this project anymore.

Anyway, thanks for your contributions!

Cheers,
Mark

**Comment by martyzz1** [comment]
Mark, apologies for completely missing your comments on this... I wasn't deliberately ignoring you, just didn't get any notification - and I finished doing the work I needed... I basically needed to port a load of tickets from github to assembla, and so I just added the functionality I needed to access the relevent assembla API endpoints.. If memory serves I think one of the changes I did fixed a small bug I found...

**Comment by markfinger** [comment]
No worries and thanks again for pushing the changes upstream.

I'll take a poke around and see if I can spot the bug you mentioned.

**Comment by martyzz1** [comment]
I think it was this commit

https://github.com/martyzz1/assembla/commit/17b801ca4e0a4bf9a3382c34f175d4340ef1faf6

On 3 October 2014 02:20, Mark Finger notifications@github.com wrote:

> No worries and thanks again for pushing the changes upstream.
> 
> I'll take a poke around and see if I can spot the bug you mentioned.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/markfinger/assembla/pull/24#issuecomment-57737580.

## 

[image: photo]
_Martin_\* Moss*
COO / CTO [image: Twitter] http://www.twitter.com/Martyzz1
m: +447734702905
w: www.sharehoods.com
w: www.migreat.com http://www.sharehoods.com/

**Pull request closed without merging**